presto-main/src/main/java/com/facebook/presto/operator/aggregation/GenericAccumulatorFactory.java

```diff
@@ -399,27 +418,136 @@ public void addInput(GroupByIdBlock groupIdsBlock, Page page)
             accumulator.addInput(groupIds, new Page(filtered.getPositionCount(), columns));
         }
 
+        @Override
+        public void evaluateFinal(int groupId, BlockBuilder output)
+        {
+            accumulator.evaluateFinal(groupId, output);
+        }
+
+        @Override
+        public void prepareFinal()
+        {
+        }
+
+        public void addBlocksToHash(GroupByIdBlock groupIds, Block[] blocks)
+        {
+            Work<Block> work = hash.markDistinctRows(new Page(blocks));
+            checkState(work.process());
+            Block distinctMask = work.getResult();
+
+            blocks[0] = distinctMask;
+            accumulator.addInput(groupIds, new Page(blocks));
+        }
+    }
+
+    private static class SpillableDistinctingGroupedAccumulator
```

Comment:
`SpillableDistinctingGroupedAccumulator` and `SpillableOrderingGroupedAccumulator` are very much the same. We just need one and call it `SpillableFinalOnlyGroupedAccumulator`